lib/superstore/persistence.rb

```diff
@@ -145,6 +144,7 @@ def update_self
       end
 
       def write(method)
+        @new_record = false
```

Comment:
our before_save callbacks wrap the `write` method. [ref](https://github.com/data-axle/superstore/blob/master/lib/superstore/callbacks.rb#L17-L19) 

This will execute for both new records and updates, but updates have already initialized `@new_record` to false, so this is harmless.